src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/Models/TrainingStatus.cs

```diff
@@ -11,62 +11,93 @@
+        /// persisted face exist in the person group. Possible values include:
+        /// 'nonstarted', 'running', 'succeeded', 'failed'</param>
+        /// <param name="created">A combined UTC date and time string that
+        /// describes person group created time.</param>
+        /// <param name="lastAction">Person group last modify time in the UTC,
+        /// could be null value when the person group is not successfully
+        /// trained.</param>
+        /// <param name="message">Show failure message when training failed
+        /// (omitted when training succeed).</param>
+        public TrainingStatus(TrainingStatusType status, System.DateTime? created = default(System.DateTime?), System.DateTime? lastAction = default(System.DateTime?), string message = default(string))
         {
-            switch( value )
-            {
-                case TrainingStatus.Nonstarted:
-                    return "nonstarted";
-                case TrainingStatus.Running:
-                    return "running";
-                case TrainingStatus.Succeeded:
-                    return "succeeded";
-                case TrainingStatus.Failed:
-                    return "failed";
-            }
-            return null;
+            Status = status;
+            Created = created;
+            LastAction = lastAction;
+            Message = message;
+            CustomInit();
         }
 
-        internal static TrainingStatus? ParseTrainingStatus(this string value)
+        /// <summary>
+        /// An initialization method that performs custom operations like setting defaults
+        /// </summary>
+        partial void CustomInit();
+
+        /// <summary>
+        /// Gets or sets training status: notstarted, running, succeeded,
+        /// failed. If the training process is waiting to perform, the status
+        /// is notstarted. If the training is ongoing, the status is running.
+        /// Status succeed means this person group is ready for Face -
+        /// Identify. Status failed is often caused by no person or no
+        /// persisted face exist in the person group. Possible values include:
+        /// 'nonstarted', 'running', 'succeeded', 'failed'
+        /// </summary>
+        [JsonProperty(PropertyName = "status")]
+        public TrainingStatusType Status { get; set; }
+
+        /// <summary>
+        /// Gets or sets a combined UTC date and time string that describes
+        /// person group created time.
+        /// </summary>
+        [JsonProperty(PropertyName = "createdDateTime")]
+        public System.DateTime? Created { get; set; }
+
+        /// <summary>
+        /// Gets or sets person group last modify time in the UTC, could be
+        /// null value when the person group is not successfully trained.
+        /// </summary>
+        [JsonProperty(PropertyName = "lastActionDateTime")]
+        public System.DateTime? LastAction { get; set; }
```

Comment:
nullable?

Reply:
Seems it is possible to have nullable `LastAction`, but I do not think we need nullable `Created`.

Reply:
Done